app/assets/javascripts/task_management_notifications.js

```diff
@@ -13,7 +13,7 @@
         userAction;
 
     userAction = $.ajax({
-      url: ("/dashboard/notifications/" + urlParam),
+      url: ("dashboard/task_management_notifications/" + urlParam),
```

Comment:
Because the url was redirecting to `notifications#update` which was also coupled to the `task_management` notifications. I moved the task management specific logic to it's controller.